src/sage/geometry/hyperplane_arrangement/ordered_arrangement.py

```diff
@@ -0,0 +1,652 @@
+            sage: L.<x, y, z> = OrderedHyperplaneArrangements(QQ)
+            sage: L(L.gens() + (x + y + z + 1,)).affine_fundamental_group().sorted_presentation()
+            Finitely presented group
+            < x0, x1, x2, x3 | x3^-1*x2^-1*x3*x2, x3^-1*x1^-1*x3*x1,
+                               x3^-1*x0^-1*x3*x0, x2^-1*x1^-1*x2*x1,
+                               x2^-1*x0^-1*x2*x0, x1^-1*x0^-1*x1*x0 >
+            sage: A = OrderedHyperplaneArrangements(QQ, names=())
+            sage: H = A(); H
+            Empty hyperplane arrangement of dimension 0
+            sage: H.affine_fundamental_group()
+            Finitely presented group <  |  >
+        """
+        K = self.base_ring()
+        if not K.is_subring(QQbar):
+            raise TypeError('the base field is not in QQbar')
+        if self._affine_fundamental_group:
+            return self._affine_fundamental_group
+        n = self.dimension()
+        r = len(self)
+        if n == 0:
+            return FreeGroup(0) / []
+        if n == 1:
+            G = FreeGroup(r) / []
+            dic = {j: G.gen(j) for j in range(r)}
+            dic[r] = [prod(G.gens()) ** -1]
+            self._affine_fundamental_group = G
+            self._affine_meridians = dic
+            return G
+        if n == 2:
+            S = self.parent().ambient_space().symmetric_space()
+            coord = vector((1,) + S.gens())
+            Af = AffinePlaneCurveArrangements(K,
+                                              names=self.parent().variable_names())
+            L = Af([vector(line.coefficients()) * coord for line in self])
+            G = L.fundamental_group()
+            self._affine_fundamental_group = G
+            self._affine_meridians = L._meridians_simpl_vertical
+            return G
+        H1 = self.hyperplane_section(proj=False)
+        G = H1.affine_fundamental_group()
+        self._affine_fundamental_group = G
+        self._affine_meridians = H1._affine_meridians
+        return G
+
+    def affine_meridians(self):
+        r"""
+        Return the meridians of each hyperplane (including the one at infinity).
+
+        OUTPUT:
+
+        A dictionary
+
+        .. NOTE::
+
+           This functionality requires the ``sirocco`` package to be installed.
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: A.<x, y> = OrderedHyperplaneArrangements(QQ)
+            sage: L = [y + x, y + x - 1]
+            sage: H = A(L)
+            sage: H.affine_meridians()
+            {0: [x0], 1: [x1], 2: [x1^-1*x0^-1]}
+            sage: L = [x, y, x + 1, y + 1, x - y]
+            sage: H = A(L)
+            sage: H.affine_meridians()
+            {0: [x4], 1: [x1], 2: [x3], 3: [x0], 4: [x2], 5: [x4^-1*x3^-1*x2^-1*x1^-1*x0^-1]}
+            sage: H = A(x, y, x + y)
+            sage: H.affine_meridians()
+            {0: [x2], 1: [x1], 2: [x0], 3: [x2^-1*x1^-1*x0^-1]}
+        """
+        if not self._affine_meridians:
```

Comment:
```suggestion
        if self._affine_meridians is None:
```
Better to be explicit as it might end up having a corner case with `bool` evaluating to `False`.